libs/blocks/quiz/utils.js

```diff
@@ -429,13 +429,21 @@ export const getAnalyticsDataForBtn = (selectedQuestion, selectedCards) => {
   return '';
 };
 
-export const getAnalyticsDataForLocalStorage = (answers) => {
+export const getAnalyticsDataForLocalStorage = (answers, umbrellaProduct, primaryProducts) => {
+  let formattedResultString = '';
   let formattedAnswerString = '';
+  if (umbrellaProduct) {
+    formattedResultString = umbrellaProduct;
+  } else {
+    primaryProducts.forEach((product) => {
```

Comment:
Any chance that `primaryProducts` is undefined or is not an array? If so, might be worth performing some checks before calling `forEach`

Reply:
+1

Reply:
Added a test case with nulls